torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -148,8 +154,10 @@ def checkpoint_wrapper(
             implementation, and is ignored if a custom ``checkpoint_fn`` is
             specified.
         offload_to_cpu (Optional[bool]):
-            Whether to offload outer activations to CPU. Note that this
-            currently only works with CheckpointImpl.REENTRANT.
+            Whether to offload activations of this wrapped module to CPU. Note
+            that if this is specified, ``checkpoint_impl`` and ``checkpoint_fn``
+            arguments will be ignored in favor of the activations being
+            offloaded to CPU. Default is ``False``.
```

Comment:
I have not been too in the loop. I am wondering why we are coupling these  `offload_to_cpu=True` and `offload_to_cpu=False` options under the same `checkpoint_wrapper()`. Why not an `activation_offload_wrapper()` and `activation_checkpoint_wrapper()` or something?

Reply:
I think we can consider the latter. This API is still in prototype stage and I wanted something that's easy to flip back and forth, but definitely agree usability wise the current options aren't what we want for a stable end product, it is too confusing with certain args only working with others and otherwise being ignored